src/scales.js

```diff
@@ -116,25 +119,34 @@ function asOrdinalType(key) {
   return registry.get(key) === position ? "point" : "ordinal";
 }
 
-export function exposeScales(figure, scaleDescriptors) {
-  const scales = figure.scales = {};
+export function exposeScales(scaleDescriptors) {
+  const scales = {};
+  const legends = [];
   for (const key in scaleDescriptors) {
     let cache;
     Object.defineProperty(scales, key, {
       enumerable: true,
       get: () => cache = cache || exposeScale(scaleDescriptors[key])
     });
+    const options = scaleDescriptors[key].options;
+    let {legend} = options;
```

Comment:
This is scoping the legend options under a legend object, which is different from how we handle axis options (which are in the same namespace as scale options, i.e., not under an axis object). I wonder whether we should flatten the legend options as well; certainly the ticks and label options seem the same as an axis, so no reason to make them different.